ipp-client.go

```diff
@@ -23,45 +18,30 @@ type Document struct {
 
 // IPPClient implements a generic ipp client
 type IPPClient struct {
-	host     string
-	port     int
 	username string
-	password string
-	useTLS   bool
-
-	client *http.Client
+	adapter  Adapter
 }
 
-// NewIPPClient creates a new generic ipp client
+// NewIPPClient creates a new generic ipp client (used HttpAdapter internally)
 func NewIPPClient(host string, port int, username, password string, useTLS bool) *IPPClient {
-	httpClient := http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		},
-	}
+	adapter := NewHttpAdapter(host, port, username, password, useTLS)
 
-	return &IPPClient{host, port, username, password, useTLS, &httpClient}
-}
-
-func (c *IPPClient) getHttpUri(namespace string, object interface{}) string {
-	proto := "http"
-	if c.useTLS {
-		proto = "https"
-	}
-
-	uri := fmt.Sprintf("%s://%s:%d", proto, c.host, c.port)
-
-	if namespace != "" {
-		uri = fmt.Sprintf("%s/%s", uri, namespace)
+	return &IPPClient{
+		username: username,
+		adapter:  adapter,
 	}
+}
 
-	if object != nil {
-		uri = fmt.Sprintf("%s/%v", uri, object)
+// NewIPPClientWithAdapter creates a new generic ipp client with given Adapter
+func NewIPPClientWithAdapter(username string, adapter Adapter) *IPPClient {
+	return &IPPClient{
+		username: username,
+		adapter:  adapter,
 	}
+}
 
-	return uri
+func (c *IPPClient) getHttpUri(namespace string, object interface{}) string {
```

Comment:
since getHttpUri is a client internal function we could remove it and use the adapter getHttpUri function directly

Reply:
fixed this, too